src/m365/spo/commands/folder/folder-retentionlabel-remove.ts

```diff
@@ -0,0 +1,163 @@
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.folderId &&
+          !validation.isValidGuid(args.options.folderId as string)) {
+          return `${args.options.folderId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['folderUrl', 'folderId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (args.options.confirm) {
+      await this.removeFolderRetentionLabel(logger, args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the retentionlabel from folder ${args.options.folderId || args.options.folderUrl} located in site ${args.options.webUrl}?`
+      });
+
+      if (result.continue) {
+        await this.removeFolderRetentionLabel(logger, args);
+      }
+    }
+  }
+
+  private async removeFolderRetentionLabel(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Removing retention label from folder ${args.options.folderId || args.options.folderUrl} in site at ${args.options.webUrl}...`);
+    }
+    try {
+      const folderProperties = await this.getFolderProperties(args);
+      const options: SpoListItemRetentionLabelRemoveCommandOptions = {
+        webUrl: args.options.webUrl,
+        listId: folderProperties.listId,
+        listItemId: folderProperties.id,
+        confirm: true,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      const spoListItemRetentionLabelRemoveCommandOutput = await Cli.executeCommandWithOutput(SpoListItemRetentionLabelRemoveCommand as Command, { options: { ...options, _: [] } });
+      if (this.verbose) {
+        logger.logToStderr(spoListItemRetentionLabelRemoveCommandOutput.stderr);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getFolderProperties(args: CommandArgs): Promise<{ id: string, listId: string }> {
+    const requestOptions: AxiosRequestConfig = {
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    let requestUrl = `${args.options.webUrl}/_api/web/`;
+
+    if (args.options.folderId) {
+      requestUrl += `GetFolderById('${args.options.folderId}')`;
+    }
+    else {
+      const serverRelativeUrl = urlUtil.getServerRelativePath(args.options.webUrl, args.options.folderUrl!);
+      requestUrl += `GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(serverRelativeUrl)}')`;
+    }
+
+    requestOptions.url = `${requestUrl}?$expand=ListItemAllFields,ListItemAllFields/ParentList&$select=ListItemAllFields/ParentList/Id,ListItemAllFields/Id`;
```

Comment:
Hi @nicodecleyre, Could you fix this part so that it also queries the ServerRelativeUrl and returns the FolderProperties object (as I commented in your `spo folder retentionlabel ensure` PR: #4320 ?

Of course you should still wait till #4348 is through..